From 82c1bd4214a14b374169159e8bd0ad65ae21d70d Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Fri, 12 Dec 2014 14:17:04 +0800
Subject: [PATCH 1104/1543] MLK-9988 PXP V4L2 output: Add YUYV format support

commit ecaca4539f622060fa0a8a4b7078976c6116fb01 from
git://git.freescale.com/imx/linux-2.6-imx.git

Add YUYV output support to pxp v4l2 output.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/media/platform/mxc/output/mxc_pxp_v4l2.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
index f71e11b..d146115 100644
--- a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
+++ b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
@@ -88,6 +88,11 @@ static struct pxp_data_format pxp_s0_formats[] = {
 		.fourcc = V4L2_PIX_FMT_UYVY,
 		.colorspace = V4L2_COLORSPACE_JPEG,
 	}, {
+		.name = "YUYV",
+		.bpp = 2,
+		.fourcc = V4L2_PIX_FMT_YUYV,
+		.colorspace = V4L2_COLORSPACE_JPEG,
+	}, {
 		.name = "Y444",
 		.bpp = 4,
 		.fourcc = V4L2_PIX_FMT_YUV444,
@@ -115,6 +120,8 @@ static unsigned int v4l2_fmt_to_pxp_fmt(u32 v4l2_pix_fmt)
 		pxp_fmt = PXP_PIX_FMT_UYVY;
 	else if (v4l2_pix_fmt == V4L2_PIX_FMT_YUV444)
 		pxp_fmt = PXP_PIX_FMT_YUV444;
+	else if (v4l2_pix_fmt == V4L2_PIX_FMT_YUYV)
+		pxp_fmt = PXP_PIX_FMT_YUYV;
 
 	return pxp_fmt;
 }
-- 
1.7.5.4

